parser.go

```diff
@@ -1507,103 +1505,145 @@ func (p *parser) getTypeSpec(pkgPath, pkgName, typeName string) (*ast.TypeSpec,
 	return astTypeSpec, true
 }
 
-func (p *parser) parseSchemaPropertiesFromStructFields(pkgPath, pkgName string, structSchema *SchemaObject, astFields []*ast.Field) {
-	if astFields == nil {
-		return
-	}
-	structSchema.Properties = orderedmap.New()
-	if structSchema.DisabledFieldNames == nil {
-		structSchema.DisabledFieldNames = map[string]struct{}{}
+func (p *parser) parseAstFields(pkgPath, pkgName string, structSchema *SchemaObject, astFields []*ast.Field) {
+	for _, astField := range astFields {
+		p.parseAstField(pkgPath, pkgName, structSchema, astField)
 	}
+}
 
-	for _, astField := range astFields {
-		fieldSchema := &SchemaObject{}
-		typeAsString := p.getTypeAsString(astField.Type)
-		typeAsString = strings.TrimLeft(typeAsString, "*")
-		isSliceOrMap := strings.HasPrefix(typeAsString, "[]") || strings.HasPrefix(typeAsString, "map[]")
-		isInterface := strings.HasPrefix(typeAsString, "interface{}")
-		if isSliceOrMap || isInterface || typeAsString == "time.Time" {
+func (p *parser) parseAstField(pkgPath, pkgName string, structSchema *SchemaObject, astField *ast.Field) {
+	fieldSchema := &SchemaObject{}
+	typeAsString := p.getTypeAsString(astField.Type)
+	isSliceOrMap := strings.HasPrefix(typeAsString, "[]") || strings.HasPrefix(typeAsString, "map[]")
+	isInterface := strings.HasPrefix(typeAsString, "interface{}")
+	if isSliceOrMap || isInterface || typeAsString == "time.Time" {
+		splitType := strings.Split(typeAsString, "]")
+		if len(splitType) > 1 {
+			valueType := splitType[1]
+			if !isBasicGoType(valueType) {
+				if _, ok := p.KnownIDSchema[valueType]; ok {
+					nestedType := p.getTypeAsString(splitType[1])
+					setNestedFieldSchemaProps(splitType[0], nestedType, fieldSchema, structSchema)
+				} else {
+					fieldSchemaObjectID, err := p.registerType(pkgPath, pkgName, typeAsString)
+					fieldSchema, err = p.parseSchemaObject(pkgPath, pkgName, typeAsString, true)
+					if err != nil {
+						p.debug(err)
+						return
+					}
+					if fieldSchemaObjectID != "" {
```

Comment:
What does it mean if the id is `""`? Does that mean an error has occurred and we should blow up? Is there a case where it can be `""` and it's OK to swallow it as we're doing here?

Reply:
when an `interface{}` gets passed in to the struct ref with error:
```2021/08/05 13:59:37 schema does not reference valid name```

Reply:
Ah, it's for `interface{}` - in which case there's nothing to ref to.